### ISSUE TITLE
MA0056 when registering EventHandlers/assinging delegates

### ISSUE THREAD

**Issue by DummkopfOfHachtenduden** [meziantou/Meziantou.Analyzer#414]
**Context**
Rule identifier: _MA0056_
Target Framework: _.NET 6.0_
Language version: _10.0_

**Describe the bug**
When registering an EventHandler or assigning a delegate with a virtual method in a constructor it is reported as MA0056 despite never actually being called. If registering event handlers or assigning delegates to virtual methods in the constructor is bonkers altogether then the wording on MA0056 should probably be changed to reflect that.

**To Reproduce**
Code or steps to reproduce the behavior:

1. Declare an event or delegate.
2. Register or assign as virtual method to it respectively inside the constructor.
3. Don't call it.

*EventHandler*
````c#
using System;

internal class Connector
{
    public event EventHandler? Connected;

    public void Connect()
    {
        // ...Imaginary socket connecting...
        this.Connected?.Invoke(this, EventArgs.Empty);
    }
}

internal class Base
{
    private readonly Connector _connector;

    public Base()
    {
        _connector = new Connector();

        // This is reported as MA0056 but it not actually be called here
        // It only referenced and thus should be compliant, shouldn't it?
        _connector.Connected += this.Connector_Connected;

        // The following code would make it non-compliant again.
        // _something.Connect();
    }

    protected virtual void Connector_Connected(object? obj, EventArgs e)
    {

    }

    public void Connect()
    {
        _connector.Connect();
    }
}

internal class Derived : Base
{
    private int value;

    public Derived()
    {
        value = 42;
    }

    protected override void Connector_Connected(object? obj, EventArgs e) => Console.WriteLine(value);
}

internal class Program
{
    public static void Main(string[] args)
    {
        var sample = new Derived();
        sample.Connect();
        Console.ReadLine();
    }
}
````

*Delegate*
````cs
using System;

internal class Connector
{
    public Action Connected;

    public void Connect()
    {
        // ...Imaginary socket connecting...
        Connected.Invoke();
    }
}

internal class Base
{
    private readonly Connector _connector;

    public Base()
    {
        _connector = new Connector()
        {
            // This is reported as MA0056 but it not actually be called here
            // It only referenced and thus should be compliant, shouldn't it?
            Connected = this.Connector_Connected,
        };

        // The following code would make it non-compliant again.
        // _something.Connect();
    }

    protected virtual void Connector_Connected()
    {

    }

    public void Connect()
    {
        _connector.Connect();
    }
}

internal class Derived : Base
{
    private int value;

    public Derived()
    {
        value = 42;
    }

    protected override void Connector_Connected() => Console.WriteLine(value);
}

internal class Program
{
    public static void Main(string[] args)
    {
        var sample = new Derived();
        sample.Connect();
        Console.ReadLine();
    }
}
````

**Comment by meziantou** [comment]
Good point! The rule implementation is too basic. I'll update it to remove some false-positives.
